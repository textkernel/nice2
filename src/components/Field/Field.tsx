import React from 'react';
import { bem } from '../../utils';
import styles from './Field.scss';

const { block, elem } = bem('Field', styles);

type Props = {
    /** Label text for the input */
    labelText: string;
    /** Component to be rendered inside the field */
    children: React.ReactNode;
};

export const Field: React.FC<Props> = ({ labelText, children }) => (
    // eslint-disable-next-line jsx-a11y/label-has-associated-control
    <label {...block()}>
        <p {...elem('label')}>{labelText}</p>
        {children}
    </label>
);

Field.displayName = 'Field';
