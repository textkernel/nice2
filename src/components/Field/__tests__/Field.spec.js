import React from 'react';
import toJson from 'enzyme-to-json';
import { Field } from '../Field';

describe('<Field> that renders an input field', () => {
    it('should render a component with a label properly', () => {
        const labelText = 'labelText';
        const wrapper = shallow(
            <Field labelText={labelText} className="customClass">
                <input />
            </Field>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('p').text()).toEqual(labelText);
    });
});
