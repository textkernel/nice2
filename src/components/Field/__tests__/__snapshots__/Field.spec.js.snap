// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field> that renders an input field should render a component with a label properly 1`] = `
<label
  className="customClass"
>
  <p
    className="Field__label"
  >
    labelText
  </p>
  <input />
</label>
`;
